src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -40,14 +40,19 @@
   public static final double DETAIL_LEVEL = 1.0;
   private static final int arrowLength = 4;
 
+  private RouteOptimizer routeOptimizer;
+
   /**
    * Draws the route to the screen.
    */
   public void drawRoute(final Graphics2D graphics, final RouteDescription routeDescription, final MapPanel mapPanel,
       final MapData mapData, final String maxMovement) {
-    if (routeDescription == null) {
+    if (routeDescription == null || mapData == null || mapPanel == null) {
```

Comment:
suspect null checks, ideally we would just not call this method if these were null. Can we enforce that on the caller side?